### PR TITLE
invidious: unstable-2024-01-29 -> 0.20.1-unstable-2024-02-18, cleanup

### PR DESCRIPTION
**Pull request by 999eagle**
## Description of changes

`lsquic` hasn't been used since https://github.com/iv-org/invidious/pull/4008. This also changes the update script to include the latest tag according to the new version conventions (see https://github.com/NixOS/nixpkgs/pull/234201).
Supersedes: #289929, #289937

CC: @GaetanLepage 

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by 999eagle** [comment]
Deployed this to my instance now

**Comment by GaetanLepage** [comment]
Thanks @999eagle ! Maybe you can include the changes from #289937 at the same time ?

**Comment by 999eagle** [comment]
Done! I didn't change `shards.nix` to sri format though because that file is generated automatically by `crystal2nix`. We probably need to update that to generate hashes in sri format.

**Review by GaetanLepage — APPROVED** [review]
LGTM ! Thanks @999eagle :)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 289943` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>invidious</li>
  </ul>
</details>

File: pkgs/servers/invidious/default.nix [hunk 1]
```diff
@@ -102,15 +91,16 @@ crystal.buildCrystalPackage rec {
...
   installCheckPhase = ''
-    INVIDIOUS_CONFIG="$(cat <<EOF
+    export INVIDIOUS_CONFIG="$(cat <<EOF
     database_url: sqlite3:///dev/null
     hmac_key: "this-is-required"
     EOF
-    )" $out/bin/invidious --help
+    )"
+    $out/bin/invidious --help
+    $out/bin/invidious --version
```
Comment (kirillrdy) [inline pkgs/servers/invidious/default.nix:100]: no need to chage, but this is usually done via https://nixos.org/manual/nixpkgs/stable/#tester-testVersion

Comment (999eagle) [reply pkgs/servers/invidious/default.nix:100]: Hmm not really. This outputs a JSON containing, among other pieces of data, a version like `2024.02.18-e8a3698`, which doesn't match the package version of `0.20.1-unstable-2024-02-18`. This was more of a debug output for myself to make sure the changes to how the version is parsed and substituted in the source code are correct.

Comment (kirillrdy) [reply pkgs/servers/invidious/default.nix:100]: Fair enough,  but you could make it try to grep for a date but it's probably not worth it

**Comment by kirillrdy** [comment]
@GrahamcOfBorg build invidious.tests

**Review by GaetanLepage — APPROVED** [review]
This seems to be in a good state now ! Thank you once again @999eagle

**Comment by kirillrdy** [comment]
@999eagle thank you !

**Pull request merged**